interfaces/acados_matlab/ocp_compile_mex.m

```diff
@@ -58,16 +62,22 @@ function ocp_compile_mex(opts)
 		end
 	else
 		if (strcmp(opts.qp_solver, 'full_condensing_qpoases'))
-			mex(mex_flags, 'CFLAGS=\$CFLAGS -std=c99 -fopenmp -DACADOS_WITH_QPOASES', acados_include, acados_interfaces_include, external_include, blasfeo_include, acados_lib_path, '-lacados', '-lhpipm', '-lblasfeo', '-lqpOASES_e', mex_files{ii})
+			mex(mex_flags, 'CFLAGS=$CFLAGS -std=c99 -fopenmp -DACADOS_WITH_QPOASES', acados_include, acados_interfaces_include, external_include, blasfeo_include, acados_lib_path, '-lacados', '-lhpipm', '-lblasfeo', '-lqpOASES_e', mex_files{ii})
 		else
-			mex(mex_flags, 'CFLAGS=\$CFLAGS -std=c99 -fopenmp', acados_include, acados_interfaces_include, external_include, blasfeo_include, acados_lib_path, '-lacados', '-lhpipm', '-lblasfeo', mex_files{ii})
-		end
-	end
+			mex(mex_flags, 'CFLAGS=$CFLAGS -std=c99 -fopenmp', acados_include, acados_interfaces_include, external_include, blasfeo_include, acados_lib_path, '-lacados', '-lhpipm', '-lblasfeo', mex_files{ii})
+    end
+  end
 end
 
 if is_octave()
-	system(['mv -f *.o build/']);
-	system(['mv -f *.mex build/']);
-else
-	system(['mv -f *.mexa64 build/']);
+  movefile('*.o', opts.output_dir)
 end
+
+for k=1:length(mex_names)
+  clear(mex_names{k})
```

Comment:
What is the `clear` for? in `sim_compile_mex` you did not add it.

Reply:
The problem is: when the mex files are loaded by Matlab you can not delete them, so the `movefile` command fails (e.g. on the second run). 
I will check this again later as i still had to do a `clear all` sometimes.